agent/data/scalar.hpp

```diff
@@ -56,30 +56,37 @@ template <typename T> class Scalar
            const std::string& iface, const std::string& prop,
            const T& initValue) :
         _value(initValue),
-        _busName(busName), _path(path), _iface(iface), _prop(prop)
+        _busName(busName), _path(path), _iface(iface), _prop(prop),
+        _onChangedMatch(sdbusplus::helper::helper::getBus(),
+                        sdbusplus::bus::match::rules::propertiesChanged(
+                            path.c_str(), iface.c_str()),
+                        std::bind(&Scalar<T>::onPropertyChanged, this,
+                                  std::placeholders::_1))
     {
-        onChangedMatch = std::make_unique<sdbusplus::bus::match::match>(
-            sdbusplus::helper::helper::getBus(),
-            sdbusplus::bus::match::rules::propertiesChanged(path.c_str(),
-                                                            iface.c_str()),
-            std::bind(&Scalar<T>::onPropertyChanged, this,
-                      std::placeholders::_1));
     }
 
     /**
      * @brief Sent request to DBus object and store new value of property
      */
     void update()
     {
-        auto r = sdbusplus::helper::helper::callMethod(
-            _busName, _path, sdbusplus::helper::PROPERTIES_IFACE, "Get", _iface,
-            _prop);
-
-        if (!r.is_method_error())
+        try
+        {
+            auto r = sdbusplus::helper::helper::callMethod(
+                _busName, _path, sdbusplus::helper::PROPERTIES_IFACE, "Get",
+                _iface, _prop);
+
+            if (!r.is_method_error())
+            {
+                value_t var;
+                r.read(var);
+                setValue(var);
+            }
+        }
+        catch (const sdbusplus::exception::SdBusError&)
         {
-            value_t var;
-            r.read(var);
-            setValue(var);
+            // Corresponded service is not started yet.
+            // When service will started, we will receive data with signal.
```

Comment:
```
// Corresponding service is not started yet
// When the service is started, we'll receive the data with a signal.
```

Are you sure though that this `catch` block needs to be empty? What's the purpose then? Just silencing the exceptions?

Reply:
yes, it is so

Reply:
The please add a comment line saying that. Something like:
```
// This catch block is just for silencing the exceptions until the service is started.
```